examples/run-transactions-complete/index.ts

```diff
@@ -0,0 +1,96 @@
+import VM from '../..'
+import Account from 'ethereumjs-account'
+import * as utils from 'ethereumjs-util'
+import { promisify } from 'util'
+import stateManager from '../../lib/state/stateManager'
+
+const Transaction = require('ethereumjs-tx') // Change when https://github.com/ethereumjs/ethereumjs-vm/pull/541 gets merged
+
+async function main() {
+  const vm = new VM()
+
+  // import the key pair
+  //   pre-generated (saves time)
+  //   used to sign transactions and generate addresses
+  const keyPair = require('./key-pair')
+
+  // Transaction to initalize the name register, in this case
+  // it will register the sending address as 'null_radix'
+  // Notes:
+  //   - A transaction has the fiels:
+  //     - nonce
+  //     - gasPrice
+  //     - gasLimit
+  //     - data
+  const rawTx1 = require('./raw-tx1')
+
+  // 2nd Transaction
+  const rawTx2 = require('./raw-tx2')
+
+  // the address we are sending from
+  const publicKeyBuf = utils.toBuffer(keyPair.publicKey)
+  const address = utils.pubToAddress(publicKeyBuf, true)
+
+  console.log('---------------------')
+  console.log('Sender address: ', utils.bufferToHex(address))
+
+  // create a new account
+  const account = new Account()
+
+  // give the account some wei.
+  account.balance = utils.toBuffer('0xf00000000000000001')
+
+  // Save the account
+  await promisify(vm.stateManager.putAccount.bind(vm.stateManager))(address, account)
+
+  // The first transaction deploys a contract
+  const createdAddress = (await runTx(vm, rawTx1, keyPair))!
+
+  // The second transaction calls that contract
+  await runTx(vm, rawTx2, keyPair)
+
+  // Now lets look at what we created. The transaction
+  // should have created a new account for the contract
+  // in the state. Lets test to see if it did.
+
+  const createdAccount = (await promisify(vm.stateManager.getAccount.bind(vm.stateManager))(
+    createdAddress,
+  )) as Account
+
+  console.log('-------results-------')
+  console.log('nonce: ' + createdAccount.nonce.toString('hex'))
```

Comment:
I'm not sure why this gives me `0x01`. It's a newly deployed contract. Shouldn't it be `0x0`?

Reply:
Hmm, this might need some extra debugging, at the end not related to the example code.

@s1na Wondered this already a couple of times: should we remove this proxy PStateManager class and directly switch to the Promise-based API for v4? Simplifies a lot and would make sense since we also switch with the other APIs directly.

Reply:
@holgerd77 We'd have to add a promise-bayed wrapper for the `merkle-patricia-tree` or use `promisify` on every invocation of its methods. Ideally we could make `StateManager` fully promise-based after `merkle-patricia-tree` has been promisified. But until then if `PStateManager` is annoying you I can pursue one of the two approaches I mentioned :)

Reply:
Ah, sorry, wasn't aware of this interdependency. No, just moving wrappers around doesn't make that much sense, then let's leave as is for now.

Reply:
I lost so much time with this, and the `1` was right 😫